src/Sentry/Internal/MainExceptionProcessor.cs

```diff
@@ -30,6 +30,68 @@ public void Process(Exception exception, SentryEvent sentryEvent)
         sentryEvent.SentryExceptions = sentryExceptions;
     }
 
+    // Sentry exceptions are sorted oldest to newest.
+    // See https://develop.sentry.dev/sdk/event-payloads/exception
+    internal IReadOnlyList<SentryException> CreateSentryExceptions(Exception exception)
+    {
+        var exceptions = WalkExceptions(exception).Reverse().ToList();
+
+        // In the case of only one exception, ExceptionId and ParentId are useless.
+        if (exceptions.Count == 1 && exceptions[0].Mechanism is { } mechanism)
+        {
+            mechanism.ExceptionId = null;
+            mechanism.ParentId = null;
+            if (mechanism.IsDefaultOrEmpty())
+            {
+                // No need to convey an empty mechanism.
+                exceptions[0].Mechanism = null;
+            }
+        }
+
+        return exceptions;
+    }
+
+    private class Counter
```

Comment:
Could `Counter` be a `struct`?

Reply:
Intentionally needs to be a class, so the same instance is passed by reference throughout all recursice calls.